plugins/pulp_rpm/plugins/migrations/0031_yum_metadata_storage_path.py

```diff
@@ -0,0 +1,39 @@
+from gettext import gettext as _
+import logging
+import os
+import tempfile
+
+from pulp.server.db import model
+from pulp_rpm.plugins.db import models
+
+
+_log = logging.getLogger('pulp')
+
+
+def migrate(*args, **kwargs):
+    """
+    Re-calculate storage path for yum_repo_metadata_file unit
+    """
+    metadata_files = models.YumMetadataFile.objects.filter()
+    repos = set()
+    for unit in metadata_files:
+        old_path = unit._storage_path
+        unit.set_storage_path(os.path.basename(unit._storage_path))
+        if unit._storage_path != old_path:
+            unit.save()
+            unit.safe_import_content(old_path)
+            repos_with_unit = model.RepositoryContentUnit.objects.filter(unit_id=unit.id)
+            repos.update(repos_with_unit)
+
+    repo_list = [repo.repo_id for repo in repos]
+    repos_to_republish = model.Distributor.objects.filter(repo_id__in=repo_list,
+                                                          last_publish__ne=None)
+    temp_dir = tempfile.gettempdir()
```

Comment:
On a cluster, it may not be clear which machine ran the migration, and thus which /tmp/ contains the file. In some deployments, for example if the migration was run in a container, its ``/tmp`` might not be available after the migrations finish running.

What do you think of dropping this file in ``/var/lib/pulp/`` instead?